graalpy/graalpy-micronaut-multithreaded/README.md

```diff
@@ -0,0 +1,507 @@
+NumPy is a native library and Python has a [Global Interpreter Lock](https://wiki.python.org/moin/GlobalInterpreterLock) that limits multithreading to use a single core at a time.
+Using GraalPy, we can lift that limitation and process requests on multiple threads using multiple CPU cores in parallel.
+![Screenshot of the app](screenshot.png)
+
+## 2. What you will need
+
+To complete this guide, you will need the following:
+
+* Some time on your hands
+* A decent text editor or IDE
+* A supported JDK[^1], preferably the latest [GraalVM JDK](https://graalvm.org/downloads/)
+
+  [^1]: Oracle JDK 17 and OpenJDK 17 are supported with interpreter only.
+  GraalVM JDK 25, Oracle JDK 21+, OpenJDK 21+ and newer with [JIT compilation](https://www.graalvm.org/latest/reference-manual/embed-languages/#runtime-optimization-support).
+
+## 3. Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the [completed example](./).
+
+## 4. Writing the application
+
+Create an application using the [Micronaut Command Line Interface](https://docs.micronaut.io/latest/guide/#cli) or with [Micronaut Launch](https://micronaut.io/launch/).
+We also recommend to use Micronaut version 4.7.4 or newer.
+You can choose to build with either Maven or Gradle.
+
+```bash
+mn create-app org.example.demo \
+--build=maven \
+--lang=java \
+--features=views-thymeleaf,graalvm \
+--test=junit
+```
+
+```bash
+mn create-app org.example.demo \
+--build=gradle_kotlin \
+--lang=java \
+--features=views-thymeleaf,graalvm \
+--test=junit
+```
+
+The `views-thymeleaf` feature will be used to render the index page, the `graalvm` feature enables the Micronaut integration with GraalVM Native Image.
+
+### 4.1. Application
+
+The generated Micronaut application will already contain the file  _Application.java_, which is used when running the application via Maven or via deployment.
+You can also run the main class directly within your IDE if it is configured correctly.
+
+`Application.java`
+```java
+public class Application {
+
+    public static void main(String[] args) {
+        Micronaut.run(Application.class, args);
+    }
+}
+```
+
+### 4.2 Dependency configuration
+
+Add the required dependencies for GraalPy in the dependency section of the POM or Gradle build script.
+
+`pom.xml`
+```xml
+<dependency>
+  <groupId>org.graalvm.python</groupId>
+  <artifactId>python</artifactId> <!-- ① -->
+  <version>24.2.0</version>
+  <type>pom</type> <!-- ② -->
+</dependency>
+<dependency>
+  <groupId>org.graalvm.python</groupId>
+  <artifactId>python-embedding</artifactId> <!-- ③ -->
+  <version>24.2.0</version>
+</dependency>
+```
+
+`build.gradle.kts`
+```kotlin
+implementation("org.graalvm.python:python:24.2.0") // ①
+implementation("org.graalvm.python:python-embedding:24.2.0") // ③
```

Comment:
these dependencies are now injected automatically when you include the graalpy plugin. In other guides I [solved](https://github.com/graalvm/graal-languages-demos/tree/main/graalpy/graalpy-javase-guide#41-dependency-configuration) this with this fomulation:
> Add the required dependencies for GraalPy in the <dependencies> section of the POM file for Maven. For Gradle, the GraalPy Gradle plugin that we will add in the next section will inject these dependencies automatically.